src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c

```diff
@@ -675,6 +675,35 @@ static int __xocl_subdev_create(xdev_handle_t xdev_hdl,
 	return retval;
 }
 
+int xocl_get_free_subdev_instance(xdev_handle_t xdev_hdl,
+	uint32_t subdev_id)
```

Comment:
Do you really need this API? if yes, please add:
  1) mark subdev as INITED to avoid the next request to get the same id.
  2) Have API to release dev_idx
  3) add corresponding create_by function 
  4) dev_idx is not instance. instance is global id and dev_idx is per card id.

Reply:
Please check. Implemented